name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/actions/DetectDuplicatesAction.java

```diff
@@ -112,7 +161,37 @@ private boolean isPotentialDuplicate(Transaction subject, Transaction other)
 
         if (!Objects.equals(other.getSecurity(), subject.getSecurity())) // NOSONAR
             return false;
+        
+        return true;
+    }
 
+    private boolean isSavingsPlanDuplicate(Transaction subject, Transaction other)
+    {
+        if (!other.getCurrencyCode().equals(subject.getCurrencyCode()))
+            return false;
+
+        if (!Objects.equals(other.getSecurity(), subject.getSecurity())) // NOSONAR
+            return false;
+
+        if (other.getAmount() != subject.getAmount())
+            return false;
+
+        LocalDateTime date = subject.getDateTime();
+        // date can be up to three days after other's date (due to shifted executions on weekends)
+        if (date.isBefore(other.getDateTime()) || date.minusDays(3).isAfter(other.getDateTime()))
+            return false;
+        
+        // number of shares might differ due to differing prices per share used by savings plan
+        // accept a difference of 10%
+        long shares = subject.getShares();
+        if (shares * 1.1 < other.getShares() || shares * 0.9 > other.getShares())
+            return false;
+        
+        // use the behavior that savings plans get a specific note and check for it
+        if (other.getNote() == null || !other.getNote().startsWith(Messages.InvestmentPlanAutoNoteLabel.substring(0, 8)))
```

Comment:
If we change the ```ImportAction#process``` methods, then let's consider adding the ```Client``` as well and checking whether the transaction is part of an InvestmentPlan (instead of comparing the comment).

Reply:
Thanks for the feedback. I didn't want to change the methods and was looking for other options to pass the values around. I will try to figure out, what is the best option to get this done, ideally without searching the investment plan twice...

Reply:
The `client` object is passed to the action during initinalization, like it is done for `InsertAction` already. Then only the investment plan transactions for the specific security are searched. Should make it more reliable and faster, too.